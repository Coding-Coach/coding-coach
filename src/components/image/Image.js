import React, { Component } from 'react';

const empty = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';

export default class Image extends Component {
  state = {
    error: false,
  };

  onError = () => {
    this.setState({
      error: true,
    });
  };

  render() {
    const { error } = this.state;
    const { alt, className, src } = this.props;

    return <img src={error ? empty : src} className={className} onError={this.onError} alt={alt} />;
  }
}
