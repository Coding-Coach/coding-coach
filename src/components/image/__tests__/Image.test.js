import React from 'react';
import { render } from 'react-testing-library';
import Image from '../Image';

describe('components/Image', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Image src="test.jpg" alt="A simple test" className="test-class" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Image', () => {
    const { container } = render(
      <Image src="test.jpg" alt="A simple test" className="test-class" />,
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.getAttribute('src')).toEqual('test.jpg');
    expect(container.firstChild.getAttribute('alt')).toEqual('A simple test');
    expect(container.firstChild.getAttribute('class')).toEqual('test-class');
  });
});
