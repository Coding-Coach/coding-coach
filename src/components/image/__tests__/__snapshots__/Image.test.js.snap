// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Image should match snapshot 1`] = `
<img
  alt="A simple test"
  class="test-class"
  src="test.jpg"
/>
`;

exports[`components/Image should render the Image 1`] = `
<img
  alt="A simple test"
  class="test-class"
  src="test.jpg"
/>
`;
